addons/payment_stripe/models/payment.py

```diff
@@ -149,14 +161,97 @@ def _get_feature_support(self):
         res['tokenize'].append('stripe')
         return res
 
+    def _handle_stripe_webhook(self, data):
+        """Process a webhook payload from Stripe.
+
+        Post-process a webhook payload to act upon the matching payment.transaction
+        record in Odoo.
+        """
+        SUPPORTED_WEBHOOKS = ['checkout.session.completed']
+        wh_type = data.get('type')
+        if wh_type not in SUPPORTED_WEBHOOKS:
+            _logger.info('unsupported webhook type %s, ignored', wh_type)
+            return False
+        else:
+            _logger.info('handling %s webhook event from stripe', wh_type)
+
```

Comment:
```py
if not self.stripe_webhook:
    raise ValidationError('webhook secret not configured')
```

Reply:
or something that fails properly

Reply:
**added this at the start of the method**:
```python
        if not self.stripe_webhook_secret:
            raise ValidationError('webhook event received but webhook secret is not configured')
```